cookbooks/kinematically_driven_subduction_2d/doc/kinematically_driven_subduction_2d.md

```diff
@@ -1,102 +1,194 @@
+boundary. This can also be seen in {numref}`fig:QQ_case1_results`.
+
+```{figure-md} fig:QQ_case1_diagnostics
+<img src="Case1_diagnostics.*" />
+
+ Case 1 diagnostic quantities of ASPECT
+```
+
+
+```{figure-md} fig:QQ_case1_results
+<img src="Case1_visc.*" style="width:60.0%" />
+
+ Case 1 viscosity snapshots at 8 and 15 My.
+```
+## Case 2a: Model with thermal evolution 
+
+Case 2a builds on Case 1 by including a heterogeneous initial temperature field 
+that combines the plate cooling model for the temperature in the plates with an 
+adiabatic temperature gradient below the plates. For the plate cooling model, 
+we assume that the overriding plate is 40 My old and the subducting plate 70 My. 
+The temperature function is too complex to comfortably write out as a function
+expression in the input file, so it is put in a new initial temperature plugin
+called ’subduction plate cooling’. This plugin needs to be compiled as a shared
+library that can be called from the input file:
+
+```{literalinclude} Case2a_temperatures.prm
+
+```
+The initial temperature distribution can be seen in  {numref}`fig:QQ_case2a_temperature_t0`
+
+```{figure-md} fig:QQ_case2a_temperature_t0
+<img src="Case2a_temp_t0.*" />
+
+ Case 2a temperature distribution at time 0
+```
+
+The thermal expansivity is kept zero in Case 2a. Therefore the thermal evolution is 
+not coupled to the mechanical evolution. A high conductivity (k = 183.33 Wm-1K-1) 
+is defined for the upper mantle to enforce the mantle adiabat in the slow convection
+system.
+
+```{literalinclude} Case2a_materialmodel.prm
+
+```
+
+We can monitor the temperature evolution, however, by tracking the
+root-mean-square (RMS) temperature and the isotherm depth using 
+two plugins called "composition_trms_statistics" and "isotherm depth".
+To this end, we add the postprocessors ’temperature statistics’ 
+and "isotherm depth" to the prm file. The RMS temperature is tracked 
+over the subducting plate (comprising BOC_SP, SHB_SP, thermal_SP) 
+and the whole domain over time. And the isotherm temperature to be 
+tracked is prescribed as 800 C.
+
+```{literalinclude} Case2a_postprocessing.prm
+
+```
+
+{numref}`fig:QQ_case2a_diagnostics` shows the RMS temperature, 
+800 C isotherm depth as well as the other diagnostics. The 800 C isotherm 
+keeps going deeper, while in the end the descend speed becomes slower. 
+The RMS temperature of the whole domain keeps decreasing due to the 
+input of cold subducting plate into model domain. The slab RMS temperature 
+cools at the initial stage, then it goes up due to being warmed up by the 
+mantle. Notice that these other diagnostics are indeed the same as for Case 1.
+
+```{figure-md} fig:QQ_case2a_diagnostics
+<img src="Case2a_diagnostics.*" />
+
+ ASPECT Case 2a and case 1, Elefant case 2a diagnostic quantities
+```
+
+## Case 2b: Model with a temperature-dependent density
+
+Case2b builds on case 2a.  Compared to Case2a, the thermal expansivity in Case 2b

```

Comment:
Please pick one way of writing out the Case names: i.e. Case2b, or Case 2b or case 2b.